### PR TITLE
Example get svid stream

### PR DESCRIPTION
**Pull request by otaviolcs3**

### CODE REVIEW

File: examples/spiffe-watcher/main.c [hunk 1]
```diff
@@ -0,0 +1,91 @@
+#include "c-spiffe/bundle/jwtbundle.h"
+#include "c-spiffe/workload/workload.h"
+#include "threads.h"
+
+int watch_X509SVIDs(void *arg)
+{
+    workloadapi_Client *client = arg;
+    /// TODO: configure watcher
```
Comment (tjamir) [inline examples/spiffe-watcher/main.c:8]: Please, remove this TODO before marking this PR as ready for merge.

File: examples/spiffe-watcher/main.c [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    err = workloadapi_Watcher_Close(watcher);
+    if(err) {
+        printf("Failed closing X.509 watcher: %u\n", err);
+    }
+    workloadapi_Watcher_Free(watcher);
+
+    return 0;
+}
+
+int watch_JWTBundles(void *arg)
```
Comment (tjamir) [inline examples/spiffe-watcher/main.c:33]: Can we reuse some code from watch_X509SVIDS?

**Review by willallves — APPROVED** [review]

File: examples/CMakeLists.txt [hunk 3]
```diff
@@ -71,3 +71,10 @@ set(SERVER_HTTP
 )
 add_executable(spiffe_http_server "${SERVER_HTTP}")
 target_link_libraries(spiffe_http_server spiffetls)
+
+# spiffe-watcher
+set(SERVER_WATCHER
+    spiffe-watcher/main.c
+)
+add_executable(spiffe_watcher "${SERVER_WATCHER}")
+target_link_libraries(spiffe_watcher client)
```
Comment (rodrigolc) [inline examples/CMakeLists.txt:80]:
```suggestion
# spiffe-watcher
set(SPIFFE_WATCHER
    spiffe-watcher/main.c
)
add_executable(spiffe_watcher "${SPIFFE_WATCHER}")
target_link_libraries(spiffe_watcher client)
```
change var name?

Comment (otaviolcs3) [reply examples/CMakeLists.txt]: Done

File: examples/spiffe-watcher/main.c [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+            }
+            putchar('\n');
+        }
+    }
+}
+
+void callback_JWTBundles(jwtbundle_Set *set, void *unused)
+{
+    if(set) {
+        jwtbundle_Bundle **bundles = jwtbundle_Set_Bundles(set);
```
Comment (rodrigolc) [inline examples/spiffe-watcher/main.c:54]:
```suggestion
        jwtbundle_Set* new_set = jwtbundle_Set_Clone(set);
        jwtbundle_Bundle **bundles = jwtbundle_Set_Bundles(new_set);
```

File: examples/spiffe-watcher/main.c [hunk 5]
```diff
@@ -0,0 +1,171 @@
...
+                printf("%s\n", str);
+            }
+
+            arrfree(str);
+            jwtutil_JWKS_Free(&jwks);
+        }
+
+        for(size_t i = 0, size = arrlenu(bundles); i < size; ++i) {
+            jwtbundle_Bundle_Free(bundles[i]);
+        }
```
Comment (rodrigolc) [inline examples/spiffe-watcher/main.c:76]:
```suggestion
        jwtbundle_Set_Free(new_set);
```
Double free error, these are already freed in workload/client.cc:828.
bundles are supposed to be copied if you need them.

alternatively, you could clone the set before getting the bundles for it (see other comment), and free the new set here

File: examples/spiffe-watcher/main.c [hunk 6]
```diff
@@ -0,0 +1,171 @@
...
+    return 0;
+}
+
+int watch_JWTBundles(void *arg)
+{
+    workloadapi_Client *client = arg;
+    workloadapi_JWTWatcherConfig config
+        = { .client = client, .client_options = NULL };
+    workloadapi_JWTCallback cb = { .args = NULL, .func = callback_JWTBundles };
+    err_t err;
```
Comment (rodrigolc) [inline examples/spiffe-watcher/main.c:114]:
```suggestion
    err_t err = NO_ERROR;
```
Valgrind complains about jump on initialized variable, this fixes it.

File: examples/spiffe-watcher/main.c [hunk 7]
```diff
@@ -0,0 +1,171 @@
...
+    }
+    workloadapi_JWTWatcher_Free(watcher);
+
+    return 0;
+}
+
+void startWatchers(void)
+{
+    thrd_t thrd_x509, thrd_jwt;
+    err_t err;
```
Comment (rodrigolc) [inline examples/spiffe-watcher/main.c:139]:
```suggestion
    err_t err = NO_ERROR;
```
Valgrind complains about jump on initialized variable, this fixes it.

File: examples/spiffe-watcher/main.c [hunk 8]
```diff
@@ -0,0 +1,171 @@
...
+}
+
+int watch_X509SVIDs(void *arg)
+{
+    workloadapi_Client *client = arg;
+    workloadapi_WatcherConfig config
+        = { .client = client, .client_options = NULL };
+    workloadapi_X509Callback cb
+        = { .args = NULL, .func = callback_X509Context };
+    err_t err;
```
Comment (rodrigolc) [inline examples/spiffe-watcher/main.c:87]:
```suggestion
    err_t err = NO_ERROR;
```
Valgrind complains about jump on initialized variable, this fixes it.

File: examples/spiffe-watcher/main.c [hunk 9]
```diff
@@ -0,0 +1,171 @@
...
+        jwtbundle_Bundle **bundles = jwtbundle_Set_Bundles(set);
+
+        for(size_t i = 0, size = arrlenu(bundles); i < size; ++i) {
+            printf("jwt bundle updated %s:\n", bundles[i]->td.name);
+
+            jwtutil_JWKS jwks
+                = { .root = NULL,
+                    .jwt_auths = jwtutil_CopyJWTAuthorities(bundles[i]->auths),
+                    .x509_auths = NULL };
+            err_t err;
```
Comment (rodrigolc) [inline examples/spiffe-watcher/main.c:63]:
```suggestion
    err_t err = NO_ERROR;
```
Valgrind complains about jump on initialized variable, this fixes it.

**Review by rodrigolc — CHANGES_REQUESTED** [review]

File: examples/spiffe-watcher/main.c (hunk 5 above)
Comment (otaviolcs3) [reply examples/spiffe-watcher/main.c]: It is enough to just free the array of bundle pointers.

**Review by rodrigolc — APPROVED** [review]

File: examples/spiffe-watcher/README.md [hunk 10]
```diff
@@ -0,0 +1,126 @@
...
+- The trust domain is `example.org`
+- The agent SPIFFE ID is `spiffe://example.org/host`.
+- There is a `spiffe-watcher` user in the system.
+
+### 1. Create the registration entry
+Create the registration entry for the spiffe-watcher workload:
+```bash
+./spire-server entry create -spiffeID spiffe://example.org/spiffe-watcher \
+                            -parentID spiffe://example.org/host \
+                            -selector unix:user:root
```
Comment (marcusvtms) [inline examples/spiffe-watcher/README.md:84]:
```suggestion
                            -selector unix:user:spiffe-watcher
```

**Pull request merged**